log/logging.go

```diff
@@ -188,22 +203,112 @@ func SetTimestampFlags(flags int) {
 	DefaultLogger.SetTimestampFlags(flags)
 }
 
+type Logger interface {
+	Fatal(description string, keysAndValues ...interface{})
+	Error(description string, keysAndValues ...interface{})
+	Warn(description string, keysAndValues ...interface{})
+	Info(description string, keysAndValues ...interface{})
+	Debug(description string, keysAndValues ...interface{})
+	Trace(description string, keysAndValues ...interface{})
+
+	SetLevel(level LogLevel)
+	SetOutput(w io.Writer)
+	SetTimestampFlags(flags int)
+	SetStaticField(name string, value interface{})
+
+	logMessage(level LogLevelName, id string, description string, keysAndValues ...interface{})
```

Comment:
To sum up what we spoke about in person: we mean this interface to allow us to change the underlying struct without breaking users of the package, not for them to have an interface they can implement. So it's fine that they can't.

As always, if they want to mock our `Logger`, they can make their own interface and pass that around, and mock it in tests. Just like we do with libraries we use that don't supply interfaces.